go/stellar/stellarsvc/service_test.go

```diff
@@ -635,6 +635,91 @@ func testRelay(t *testing.T, yank bool) {
 	require.Equal(t, "Payment already claimed by "+tcs[claimant].Fu.Username, err.Error())
 }
 
+func TestRelayResetClaim(t *testing.T) {
+	testRelayReset(t, false)
+}
+
+func TestRelayResetYank(t *testing.T) {
+	testRelayReset(t, true)
+}
+
+func testRelayReset(t *testing.T, yank bool) {
+	tcs, cleanup := setupTestsWithSettings(t, []usetting{usettingFull, usettingFull})
+	defer cleanup()
+
+	acceptDisclaimer(tcs[0])
+
+	tcs[0].Backend.ImportAccountsForUser(tcs[0])
+	tcs[0].Backend.Gift(getPrimaryAccountID(tcs[0]), "10")
+
+	sendRes, err := tcs[0].Srv.SendCLILocal(context.Background(), stellar1.SendCLILocalArg{
+		Recipient: tcs[1].Fu.Username,
+		Amount:    "4",
+		Asset:     stellar1.AssetNative(),
+	})
+	require.NoError(t, err)
+
+	details, err := tcs[0].Backend.PaymentDetails(context.Background(), tcs[0], sendRes.KbTxID.String())
+	require.NoError(t, err)
+
+	typ, err := details.Summary.Typ()
+	require.NoError(t, err)
+	require.Equal(t, stellar1.PaymentSummaryType_RELAY, typ)
+
+	// Reset and reprovision
+	kbtest.ResetAccount(tcs[1].TestContext, tcs[1].Fu)
+	require.NoError(t, tcs[1].Fu.Login(tcs[1].G))
+
+	teamID := details.Summary.Relay().TeamID
+	t.Logf("Team ID is: %s", teamID)
+
+	var claimant int
+	if !yank {
+		// Admit back to the team.
+		err = teams.ReAddMemberAfterReset(context.Background(), tcs[0].G, teamID, tcs[1].Fu.Username)
+		require.NoError(t, err)
+
+		acceptDisclaimer(tcs[1])
+		tcs[1].Backend.ImportAccountsForUser(tcs[1])
+
+		claimant = 1
+	} else {
+		// User0 will try to claim the funds back without readding user1 to the
+		// impteam. Also do not accept disclaimer as user1.
+		claimant = 0
+	}
+
+	history, err := tcs[claimant].Srv.RecentPaymentsCLILocal(context.Background(), nil)
+	require.NoError(t, err)
+	require.Len(t, history, 1)
```

Comment:
If the claimant is user 0 I'd expect it to see in its history the gift. I guess this is an inaccuracy in the mock. You could only look at the last payment in the list to be future proof against someone fixing the mock. I don't feel strongly.